### ISSUE TITLE
508-defect-3: "Finish this request later" button should be styled as a button

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#25111]
# [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

<hr/>

## Point of Contact

**VFS Point of Contact:** Josh Kim

## Details

The "Finish this request later" button is styled as a link which is materially dishonest causing several issues:
- voice interface users may not be able to activate the button should they say "click on finish this request later link"
- screen reader users may be confused on help calls if they are instructed to click on a link when the screen reader announces it as a button
- low vision screen reader users may be confused that the screen reader is announcing a visually styled link as a button
- breaks consistency with other buttons across the interface

For more on this long-tail effort, check out our [buttons vs. links Mural board](https://app.mural.co/t/vsa8243/m/vsa8243/1610052905994/253723ef4b2ee9a726f538876a257df94d468db1) and this [Slack thread discussing _why_ material honesty is so important](https://dsva.slack.com/archives/C8E985R32/p1610659601009200?thread_ts=1610652897.002600&cid=C8E985R32).

"A few years at an accessibility event, a blind man was telling a story about how he was on the phone with someone who was trying to help him sign up for a service. The person on the phone told him to click the “sign up button” and he couldn’t find it because it was a link. He told the story with a lot more gravity than I can, but that was the point where I realized how important the human-human communication part of accessibility is, it’s not just machine to human." - Tim Wright

## Acceptance Criteria

- [ ] "Finish this request later" button is styled as a button

## Environment

* Operating System: all
* Browser: any
* Screenreading device: any
* Server destination: staging

## Solution 

Style the link as a button. If there is concern about this taking away attention from the "Back" and "Continue" buttons, a more long term UX solution should be explored. For example, "Back" is semantically more of a _link_ because it's not submitting anything, it's just moving the user backwards to the previous page. If that's placed at the top of the form and not the bottom (which is more resourceful to screen reader users), more space will be freed up at the bottom.

On that note, the "finish this request later" button _should not_ follow the "Continue" button as screen reader users will _not_ expect that more content exists below it. A more useful location may higher up in the page before the form's contents actually begin. That way a screen reader user, after hitting "Continue" on a previous page will first hear: "Step x of y: lorem ipsum" -> "Back, link" -> "Finish this request later, button" -> "Question"

## WCAG or Vendor Guidance (optional)

* [Adam Silver: But sometimes buttons look like links](https://adamsilver.io/articles/but-sometimes-buttons-look-like-links/)
* [VA Design System guidance on buttons](https://design.va.gov/components/buttons#guidance)

## Screenshots or Trace Logs

<img width="1788" alt="Screen Shot 2021-05-24 at 7 27 31 AM" src="https://user-images.githubusercontent.com/14154792/119342489-2694c400-bc63-11eb-9e11-7b7e2ece93da.png">

**Comment by caw310** [comment]
@joshkimux , this issue was discussed today in the DSC meeting. Here is the follow. 

- Recode it as a link instead of a button so materially honest.
- Place above Back and Continue buttons. 
- Recommended as a temporary solution.

**Comment by Mottie** [comment]
Temporary fix that moves the "button" above the back & continue buttons has been merged. Still pending is (from [this comment](https://github.com/department-of-veterans-affairs/vets-website/pull/17548#discussion_r648764082))

>  three changes might help:
> 1. Changing this to a `Link` from `react-router-dom` to preserve Redux state when navigating to a new URL
> 2. Make this link _only_ navigate to a new page (not call a function which both saves and navigates)
> 3. Make the new page do the saving (which is why we needed to preserve the Redux state)
>     - This will require a design change (which is why I mentioned the scope creep :sweat_smile:)

**Comment by Mottie** [comment]
I've merged in another change that renders the "Finish this request later" link above the submit button on the review & submit page.

**Comment by Mottie** [comment]
`"Finish this request later" button is styled as a button` change to be addressed by the Design system team

**Comment by SarahKay8** [comment]
@caw310 @caw310 Hi Carol, per Josh's recommendations, "Finish this request later" button has successfully been placed over "Continue + Back" buttons however "Finish this request later" button is still styled as link  

Screenshot for reference: ![Image 7-18-22 at 2.36 PM.jpg](https://images.zenhubusercontent.com/623cd4d6774110eeb19922cb/463c4ae8-1547-433b-9cff-2b1994c94103)

**Comment by k80bowman** [comment]
Thank you @SarahKay8 and @joshkimux. As this is a forms library issue, I've added the forms label. @taylorkaren @micahchiang how would you like to handle this?

**Comment by taylorkaren** [comment]
Remaining work to do: The "Finish this request later" button is styled as a link but should be implemented as a button.

**Comment by obliviga** [comment]
Hi @joshkimux, should the "Finish this request later" button be styled as a primary or secondary button? Thanks!

Reference: https://design.va.gov/components/button/#examples

**Comment by joshkimux** [comment]
@obliviga this is tricky, because styling this as a primary or secondary button may negatively impact the sighted user experience for mobile users or folks with cognitive disabilities. I think it would be helpful to have a more upstream conversation about the questions I raised earlier. It may be easier to keep this as a `defect-level-3` in the old (current) form system and attempt to resolve this officially in the new form system:

- does this need to be placed in this particular location?
- what are the rules around finishing something later?
- if forms are automatically saved, could folks finish this request later at anytime and simply leave and return to the form
- could that be communicated via better designed content?
- etc

**Comment by joshkimux** [comment]
Hey folks, quick update here. [Following this thread](https://dsva.slack.com/archives/C023ZKDJJK1/p1659106873144419) with @thehastingsp and @davidakennedy , I'm curious if we might be able to:
- look into the current usage of this button, how many Veterans are currently using it?
- review past research on this button, has the need for it changed?
- test what it might look like _without_ this button in place of well written content (@DanielleThierryUSDSVA for awareness)

**Comment by joshkimux** [comment]
@jeana-adhoc I'm not sure if we may have talked about this, but bringing this up just in case it's relevant to your work on the forms team. The last comment has our most recent context on the issue.

**Comment by rsmithadhoc** [comment]
@caw310 I believe this one (and https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1766) would be up to the forms team to address, as this isn't in the design system.

**Comment by caw310** [comment]
@tbaker1026 , would the VFFT be able to pick this up?

**Comment by caw310** [comment]
@tbaker1026 Would you add this to your backlog since it is a form issue?

**Comment by jeana-adhoc** [comment]
@caw310 - This might be a question for @humancompanion-usds. We just tested moving this even lower on a page, and it tested fine. 

How do you handle competing old tickets with new information on the DST?

**Comment by caw310** [comment]
@jeana-adhoc We are actually closing old tickets and asking people to open a new ticket if the issue persists with the new VADS components.

**Comment by jeana-adhoc** [comment]
Thanks, @caw310 - We might do the same here. But I need to get my head around what the remaining issue is, and find out if its still an issue.

**Comment by humancompanion-usds** [comment]
@jeana-adhoc and @tbaker1026 - At your leisure, move this over to the forms team repo, or open a new issue. Jeana - Sounds like the remaining work is for us to update our forms templates in Figma with the new location, update it in code, and then socialize that change?
